### ISSUE TITLE
Add button to replay test runs

### ISSUE THREAD

**Issue by PetarKirov** [CyberShadow/DAutoTest#3]
There are cases where PR(s) from one repository depend on PR(s) in other repositories. Usually, whenever a PR is merged the auto-tester deprecates the test runs for all open PRs replays them. If such functionality is hard to implement or too resource consuming, I suggest adding a button to the UI which contributors could use to manually restart the testing for a particular PR. The Jenkins project tester also has the option for manual restart of tests, which I find very handy.

**Comment by CyberShadow** [comment]
DAutoTest will retest all PRs after a change in the target branch, so this is not needed.

**Comment by CyberShadow** [comment]
A "Retry" button would be good to retest network failures, though.

**Comment by PetarKirov** [comment]
> DAutoTest will retest all PRs after a change in the target branch, so this is not needed.

By target branch you mean a branch in the dlang [meta repo](https://bitbucket.org/cybershadow/d/src), or just a branch in the repo on which the PR was opened? Because my question is specifically about retesting PRs to one repo on changes to another one. E.g. retesting all druntime PRs when a phobos PR was merged.

**Comment by CyberShadow** [comment]
Yes, meta repo. Merged PRs on any D repo will update the meta-repo branch.

**Comment by PetarKirov** [comment]
Thanks for the clarification. A related features request is to display the status of the test runner - i.e. how many PRs need to be tested before the current PR would be.

**Comment by CyberShadow** [comment]
The queue is re-evaluated every time before a PR is tested. The current algorithm is to test all PRs whose tip commit SHA1s have never been tested, in reverse chronological order, followed by all PRs whose tip commit SHA1s + target branch SHA1 combination have never been tested, also in reverse chronological order.

**Comment by WebFreak001** [comment]
this seems to have occurred on https://github.com/dlang/dmd/pull/11016 so I think this would be a good addition to have. Could also just be an undocumented http request like it is for clearing documentation cache for dpldocs right now.

**Comment by Geod24** [comment]
Yes, I've seen it happen several times, and it's always a pain to rebase just to retrigger the doc tester, because it also retriggers all the other CIs. And for contributors without write access, it needs they will loose the `auto-merge` label too.

**Comment by CyberShadow** [comment]
Yes, sorry for not addressing this problem for so long. The reason for this is that I've been flip-flopping between improving this project and working on its successor, which also integrates https://perf.dlang.io/ automatic performance regression testing.
